sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,260 @@
 pragma solidity ^0.4.19;
 
 contract SMC {
-  event TxToShard(address indexed to, int indexed shardId, int receiptId);
-  event CollationAdded(int indexed shardId, uint expectedPeriodNumber, 
-                     bytes32 periodStartPrevHash, bytes32 parentHash,
-                     bytes32 transactionRoot, address coinbase,
-                     bytes32 stateRoot, bytes32 receiptRoot,
-                     int number, bool isNewHead, int score);
-  event Deposit(address collator, int index);
-  event Withdraw(int index);
-
-  struct Collator {
-    // Amount of wei the collator holds
-    uint deposit;
-    // The collator's address
-    address addr;
+  event HeaderAdded(uint indexed shard_id, bytes32 parent_hash,
+                    bytes32 chunk_root, int128 period, int128 height,
+                    address proposer_address, uint proposer_bid,
+                    bytes proposer_signature);
+  event NotaryRegistered(address notary, uint pool_index);
+  event NotaryDeregistered(address notary, uint pool_index, uint deregistered_period);
+  event NotaryReleased(address notary, uint pool_index);
+  event ProposerRegistered(uint pool_index);
+  event ProposerDeregistered(uint index);
+  event ProposerReleased(uint index);
+
+  struct Notary {
+    uint deregisteredPeriod;
+    uint poolIndex;
+    bool deposited;
   }
-
-  struct CollationHeader {
-    bytes32 parentHash;
-    int score;
+  
+  struct Proposer {
+    uint deregisteredPeriod;
+    uint balances
   }
 
-  struct Receipt {
-    int shardId;
-    uint txStartgas;
-    uint txGasprice;
-    uint value;
-    bytes32 data;
-    address sender;
-    address to;
+  struct CollationHeader {
+    uint shardId;             // Number of the shard ID
+    bytes32 parentHash;       // Hash of the parent collation
+    bytes32 chunkRoot;        // Root hash of the collation body
+    uint period;              // Period which header should be included
+    uint height;              // Height of the collation
```

Comment:
Can you remove this field? If we need it, please elaborate on what it means (the tree height since genesis)